### ISSUE TITLE
No rerendering when URL parameter changes

### ISSUE THREAD

**Issue by mmontone** [40ants/reblocks-navigation-widget#4]
This test (if paths are equal) is not enough for my use case because I'd like to rerender when an URL parameter changes, not only when path changes:

https://github.com/40ants/reblocks-navigation-widget/blob/2f67710c5da61bbf80f85932aa5e191a435de0ac/src/core.lisp#L197

**Comment by svetlyak40wt** [comment]
Navigation widget only works with path not with GET parameters. Thus, if you want, you can encode some parameters as part of the URL.

Right now it is not convenient to extract such "parameters" out from the path, but I have an experimental code which allows to define routes like this:

```lisp
(defroutes routes
  (("/invite/:code")
   (make-invite-page code))
  (("/school/:id" :id #'parse-integer)
   (make-school-page :school-id id))
  (("/school/:id/play" :id #'parse-integer)
   (make-school-game-page :school-id id))
  ("/" (make-top-decks-page)))
```

If you are interested in testing, I can cleanup the code and upload this experimental branch to the github.

**Comment by mmontone** [comment]
Yes. I think I may be wrong with what I expect from the navigation widget. So I should give my implementation another thought. 

That parsing of urls looks interesting, and could a nice feature to have. I think you could make that optional, like, if a path string is passed, then you don't parse arguments, and if a list with path + args spec, then you do the parsing. And you don't need to work on another branch, as the new parsing would be backwards-compatible.

Thank you.
